### PR TITLE
[Feature] add kube-router (alternative CNI-Plugin that can also replace kube-proxy)

### PR DESCRIPTION
**Pull request by iwilltry42**
This PR adds the template source for the `kube-router` CNI-Plugin, which can also run a service proxy replacing kube-proxy.

Related PRs:
- ~https://github.com/rancher/types/pull/1058~ (dropped, because types are now included in rancher/rke)
- https://github.com/rancher/rke/pull/1831

### CODE REVIEW

File: rke/k8s_rke_system_images.go [hunk 1]
```diff
@@ -2307,6 +2307,7 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
 			CanalFlexVol:              m("quay.io/calico/pod2daemon-flexvol:v3.8.1"),
 			WeaveNode:                 m("weaveworks/weave-kube:2.5.2"),
 			WeaveCNI:                  m("weaveworks/weave-npc:2.5.2"),
+			KubeRouterCNI:             m("cloudnativelabs/kube-router:v1.0.0"),
```
Comment (superseb) [inline rke/k8s_rke_system_images.go]: Can we update this to latest (1.2.2)?

File: rke/templates/kube-router.go [hunk 2]
```diff
@@ -0,0 +1,479 @@
...
+            mv ${TMP} /etc/cni/net.d/10-kuberouter.conflist;
+          fi;
+          if [ ! -f /var/lib/kube-router/kubeconfig ]; then
+            TMP=/var/lib/kube-router/.tmp-kubeconfig;
+            cp /etc/kube-router/kubeconfig ${TMP};
+            mv -f ${TMP} /var/lib/kube-router/kubeconfig;
+          fi;
+          mkdir -p /opt/cni/bin;
+          if [ ! "$(ls -A /opt/cni/bin)" ]; then
+            wget https://github.com/containernetworking/plugins/releases/download/v0.8.3/cni-plugins-linux-amd64-v0.8.3.tgz -O /tmp/cni-plugins-linux-amd64-v0.8.3.tgz && tar -xf /tmp/cni-plugins-linux-amd64-v0.8.3.tgz -C /opt/cni/bin/;
```
Comment (superseb) [inline rke/templates/kube-router.go]: This is provided by rke-tools, why this fallback (which will also break airgap but in general, what is the fallback good for as if binaries are not provided it should just not work)

File: rke/templates/kube-router.go [hunk 3]
```diff
@@ -0,0 +1,479 @@
...
+          name: cni-bin-dir
+        - mountPath: /etc/kube-router
+          name: kube-router-cfg
+        - name: kubeconfig
+          mountPath: /var/lib/kube-router
+      hostNetwork: true
+      dnsPolicy: ClusterFirstWithHostNet
+      dnsConfig:
+        nameservers:
+          - 1.1.1.1
```
Comment (superseb) [inline rke/templates/kube-router.go]: Possible airgap issue?

**Comment by superseb** [comment]
* Can you share the upstream template used for this?
* If this goes in, it will be in k8s v1.21, can you update the code for this? (needs to go in `dev-v2.6`)

File: rke/templates/kube-router.go (hunk 2 above)
Comment (iwilltry42) [reply rke/templates/kube-router.go]: Well.. I didn't know that it was included in rke-tools :thinking: 
Maybe back when I started this PR, it wasn't as I think I copied most things from existing CNI-Plugins. Anyway, if it's included there, I'm going to remove it here :+1:

File: rke/k8s_rke_system_images.go (hunk 1 above)
Comment (iwilltry42) [reply rke/k8s_rke_system_images.go]: :heavy_check_mark:

File: rke/templates/kube-router.go (hunk 3 above)
Comment (iwilltry42) [reply rke/templates/kube-router.go]: removed as I updated the whole manifest based on upstream

**Comment by iwilltry42** [comment]
@superseb 

>     * Can you share the upstream template used for this?

It's based on https://github.com/cloudnativelabs/kube-router/blob/4e13a1db620c0e3c22989132c52eb2f0c401965f/daemonset/generic-kuberouter-all-features.yaml (updated to this commit just now)
 
>     * If this goes in, it will be in k8s v1.21, can you update the code for this? (needs to go in `dev-v2.6`)

Undid the accidental merge of master, rebased the PR and merged in dev-v2.6 :+1:

**Pull request closed without merging**